### PR TITLE
Add support for SkeletonNumPad

### PR DESCRIPTION
**Pull request by skeletonkbd**
## Description

Add support for SkeletonNumPad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Add support for SkeletonNumPad

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/skeletonkbd/skeletonnumpad/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            if (record->event.pressed) {
+                // when keycode QMKURL is pressed
+                SEND_STRING("https://qmk.fm/\n");
+            } else {
+                // when keycode QMKURL is released
+            }
+            break;
+    }
+    return true;
+}
```
Comment (zvecr) [inline keyboards/skeletonkbd/skeletonnumpad/keymaps/default/keymap.c:80]:
```suggestion
```

File: keyboards/skeletonkbd/skeletonnumpad/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    _L2,
+    _L3
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    QMKBEST = SAFE_RANGE,
+    QMKURL
+};
+
```
Comment (zvecr) [inline keyboards/skeletonkbd/skeletonnumpad/keymaps/default/keymap.c:31]:
```suggestion
```

File: keyboards/skeletonkbd/skeletonnumpad/config.h [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+
```
Comment (zvecr) [inline keyboards/skeletonkbd/skeletonnumpad/config.h:149]:
```suggestion
```

File: keyboards/skeletonkbd/skeletonnumpad/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    _L2,
+    _L3
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    QMKBEST = SAFE_RANGE,
+    QMKURL
+};
+
```
Comment (zvecr) [inline keyboards/skeletonkbd/skeletonnumpad/keymaps/via/keymap.c:31]:
```suggestion
```

File: keyboards/skeletonkbd/skeletonnumpad/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            if (record->event.pressed) {
+                // when keycode QMKURL is pressed
+                SEND_STRING("https://qmk.fm/\n");
+            } else {
+                // when keycode QMKURL is released
+            }
+            break;
+    }
+    return true;
+}
```
Comment (zvecr) [inline keyboards/skeletonkbd/skeletonnumpad/keymaps/via/keymap.c:80]:
```suggestion
```

File: keyboards/skeletonkbd/skeletonnumpad/rules.mk [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (zvecr) [inline keyboards/skeletonkbd/skeletonnumpad/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
```

File: keyboards/skeletonkbd/skeletonnumpad/readme.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+-   Hardware Supported: SkeletonNumPad
+-   Hardware Availability: https://github.com/skeletonkbd/SkeletonNumPad
+
+Make example for this keyboard (after setting up your build environment):
+
+    make skeletonnumpad:default
+
+Flashing example for this keyboard:
+
+    make skeletonnumpad:default:flash
```
Comment (zvecr) [inline keyboards/skeletonkbd/skeletonnumpad/readme.md:17]:
```suggestion
    make skeletonkbd/skeletonnumpad:default

Flashing example for this keyboard:

    make skeletonkbd/skeletonnumpad:default:flash
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/skeletonkbd/skeletonnumpad/config.h [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+/*== or choose animations ==*/
+//#    define RGBLIGHT_EFFECT_BREATHING
+//#    define RGBLIGHT_EFFECT_RAINBOW_MOOD
+//#    define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+//#    define RGBLIGHT_EFFECT_SNAKE
+//#    define RGBLIGHT_EFFECT_KNIGHT
+//#    define RGBLIGHT_EFFECT_CHRISTMAS
+//#    define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//#    define RGBLIGHT_EFFECT_RGB_TEST
+//#    define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline keyboards/skeletonkbd/skeletonnumpad/config.h:84]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/skeletonkbd/skeletonnumpad/rules.mk [hunk 9]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/skeletonkbd/skeletonnumpad/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/skeletonkbd/skeletonnumpad/rules.mk [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/skeletonkbd/skeletonnumpad/rules.mk:21]:
```suggestion
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```

File: keyboards/skeletonkbd/skeletonnumpad/skeletonnumpad.h [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+
+#define LAYOUT( \
+	K00, K01, K02, K03, \
+	K10, K11, K12, K13, \
+	K20, K21, K22,      \
+	K30, K31, K32, K33, \
+	K40,      K42  \
```
Comment (drashna) [inline keyboards/skeletonkbd/skeletonnumpad/skeletonnumpad.h:35]:
```suggestion
#define LAYOUT_numpad_5x4( \
	K00, K01, K02, K03, \
	K10, K11, K12,      \
	K20, K21, K22, K13, \
	K30, K31, K32,      \
	K40,      K42, K33  \
```

File: keyboards/skeletonkbd/skeletonnumpad/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+		KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+		KC_P7, KC_P8, KC_P9, KC_PPLS,
+		KC_P4, KC_P5, KC_P6,
+		KC_P1, KC_P2, KC_P3, KC_PENT,
+		KC_P0, KC_PDOT),
```
Comment (drashna) [inline keyboards/skeletonkbd/skeletonnumpad/keymaps/default/keymap.c:33]:
```suggestion
    [_BASE] = LAYOUT_numpad_5x4(
        KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,   KC_P8,   KC_P9,
        KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
        KC_P1,   KC_P2,   KC_P3,
        KC_P0,   KC_PDOT,          KC_PENT
    ),
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/skeletonkbd/skeletonnumpad/rules.mk [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/skeletonkbd/skeletonnumpad/rules.mk:15]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**